packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -812,12 +812,17 @@ class ColorScheme with Diagnosticable {
       'This feature was deprecated after v3.18.0-0.1.pre.',
     )
     Color? surfaceVariant,
-  }) : _primaryContainer = primaryContainer,
+  }) : primary = _lightSystemPalette?.canvasText.value ?? primary ?? const Color(0xff0000ba),
+       onPrimary = _lightSystemPalette?.canvas.value ?? onPrimary ?? Colors.white,
+       _primaryContainer = primaryContainer,
        _onPrimaryContainer = onPrimaryContainer,
        _primaryFixed = primaryFixed,
        _primaryFixedDim = primaryFixedDim,
        _onPrimaryFixed = onPrimaryFixed,
        _onPrimaryFixedVariant = onPrimaryFixedVariant,
+       secondary =
```

Comment:
In ColorScheme, onXXX is usually used for foreground and XXX is for background. So I think accentColorText should be mapped to **on**Secondary, and accentColor is for secondary.

Reply:
So you are saying the mapping should be:
```
primary: _lightSystemPalette?.canvas.value
onPrimary: _lightSystemPalette?.canvasText.value

secondary: _lightSystemPalette?.accentColor.value
onSecondary: _lightSystemPalette?.accentColorText.value

surface: _lightSystemPalette?.canvas.value
onSurface: _lightSystemPalette?.canvasText.value
```
right?

Looks like I got some of them reversed in the PR :) I'll fix it.

Reply:
Yes! All the text-related color should be mapped to onXXX colors:)